packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -750,6 +760,50 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
     SystemNavigator.pop();
   }
 
+  @protected
+  @visibleForTesting
+  Future<void> handleStartBackGesture(Map<dynamic, dynamic> arguments) async {
+    final AndroidBackEvent backEvent = AndroidBackEvent.fromMap(arguments);
+    for (final WidgetsBindingObserver observer in List<WidgetsBindingObserver>.of(_observers)) {
+      if (await observer.startBackGesture(backEvent)) {
+        return;
```

Comment:
You are calling addObserver in initState of AndroidBackGestureDetector. Is it guaranteed that the observer that you want will be called here?

I wonder if there is a better way for this use case. Maybe by finding the current route of the navigator or something?

handlePopRoute (above) also suffers from this, though, so this is an existing problem, and I guess it's acceptable if there is no easy way around it right now.